pkg/runtime/wfengine/backends/actors/workflow_actor.go

```diff
@@ -635,11 +649,24 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 				// Setting executionStatus to failed if workflow has failed/terminated/cancelled
 				executionStatus = diag.StatusFailed
 			}
+			wfExecutionElapsedTime = diag.ElapsedSince(state.workflowStartTime)
```

Comment:
This needs to be set at other places as well e.g. when workflow is returning with `recoverable error` or when workflow is completing successfully.

Reply:
As discussed offline, this is the only place we need to calculate wfExecutionElapsedTime.